Feature 2925 update contributions docs (#2932)
mercedes-benz/sechub: sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc

[[section-documentation-markdown]]
==== Markdown
Simple technical information -e.g. about the content of an folder

> - will be written in a README.md with markdown syntax.

Reviewer: ```suggestion
- should be written in a README.md with markdown syntax.
```

Author: changed

See https://en.wikipedia.org/wiki/Markdown for `markdown` definition

[[section-documentation-asciidoc]]
==== Asciidoc

> When ever writing a full scaled documentation - we do this with Asciidoc. +

Reviewer: ```suggestion
When a full scaled documentation is needed it should be done with Asciidoc. +
```

[[section-documentation-usecase-event-overview]]
===== Usecase event overview diagrams

> For some usecases we got event trace information (from special integration tests). With those informations

Reviewer: ```suggestion
For some usecases we got event trace information (from special integration tests). With this information
```

> additional diagrams are generated and included at Usecase description level. They show the events happening

Reviewer: ```suggestion
additional diagrams are generated and included at usecase description level. They show the events happening
```
